### PR TITLE
Update Windows images to RS5

### PR DESCRIPTION
**Pull request by damonbarry**
Update the Dockerfiles to point to the RS5 nanoserver base image.

This change also makes some minor tweaks to the Windows installer script in preparation for turning on UDS/RS5:
- If an archive folder is passed rather than a ZIP file, create the iotedge directory first before trying to copy to it. Not sure why this wasn't failing before, maybe the build job that uses this feature is pre-creating the folder.
- When creating the parent directory for the UNIX domain socket files, pipe to null so that the `Get-SecurityDaemon` function _only_ returns the intended boolean value.
- During uninstall, if the event log component directory doesn't exist (e.g., because it was already deleted), ensure `Remove-SecurityDaemonResources` still returns `$true` so that the user sees a satisfying completion message ("Successfully uninstalled IoT Edge.") at the end.
- Print some more errors when the -Verbose switch is specified.
- During uninstall, remove the `IOTEDGE_HOST` system environment variable.
- Use PowerShell's `Get-Acl` and `Set-Acl` instead of .NET Core's `[System.IO.Directory]::GetAccessControl` and `[System.IO.Directory]::SetAccessControl` (which Windows IoT Core doesn't have).
- Set up the service to automatically restart when it fails.

### CODE REVIEW

**Comment by veyalla** [comment]
Before merging this we need to make the call that we won't support RS4 Hosts because RS5 containers cannot run on RS4, even with Hyper-V isolation.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 1]
```diff
@@ -359,19 +360,23 @@ function Remove-SecurityDaemonResources {
...
     }
     elseif ($CmdErr.FullyQualifiedErrorId -ne "PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand") {
+        Write-Verbose "$CmdErr"
         Write-Warning "Could not delete '$EdgeEventLogMessagesPath'."
         Write-Warning "If you're reinstalling or updating IoT Edge, then this is safe to ignore."
         Write-Warning ("Otherwise, please close Event Viewer, or any PowerShell windows where you ran Get-WinEvent, " +
             "then run Uninstall-SecurityDaemon again with '-Force'.")
     }
     else {
-        $success = $false
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: This is still needed.

Comment (damonbarry) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: With that statement in place, if iotedged is _mostly but not entirely_ uninstalled on a user's system and they call `Uninstall-SecurityDaemon -Force`, then `Remove-SecurityDaemonResources` would return `$false` and the uninstall operation would effectively be considered a failure, even if it managed to completely uninstall the remaining bits of iotedged.

Why is it needed?

**Comment by damonbarry** [comment]
@veyalla I won't merge until we're sure we have everything else in place.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 (hunk 1 above)
Comment (arsing) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: As discussed offline, yes you're right. I broke this in a135bdfb35fc3a163ebeaf7cb211052fa0410a16; it should not set `$success` to `$false` in that branch.

`Uninstall-SecurityDaemon` doesn't print any error when `Remove-SecurityDaemonResources` returns `$false`; it just doesn't print the success message. That's probably why I didn't notice it when I broke it...

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 2]
```diff
@@ -861,5 +866,22 @@ function Write-HostRed {
     Write-Host -ForegroundColor Red @args
 }
 
+function Remove-Permissions {
+    $Path = $args
+    $User = 'BUILTIN\Users'
+    Write-HostGreen "Remove $User permission to $Path"
+    
+    # Remove inherited Write permission for BUILTIN\Users
+    icacls $Path /inheritance:d | Out-Null
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `Invoke-Native`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 3]
```diff
@@ -861,5 +866,22 @@ function Write-HostRed {
...
+    $Acl = [System.IO.Directory]::GetAccessControl($Path)
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `$acl` (lower-case A)

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 4]
```diff
@@ -861,5 +866,22 @@ function Write-HostRed {
     Write-Host -ForegroundColor Red @args
 }
 
+function Remove-Permissions {
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `function Remove-Permissions([string] $Path) {`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 5]
```diff
@@ -861,5 +866,22 @@ function Write-HostRed {
...
+    $Path = $args
+    $User = 'BUILTIN\Users'
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `$user` (lower-case U)

**Pull request merged**